controllers/profiles/getOtherProfileByUsername.js

```diff
@@ -32,7 +48,7 @@ module.exports = async (req, res) => {
     });
 
     const isFollowing = await sequelize.query(isFollowingQuery, {
-      replacements: {user_id_follower: req?.userId, user_id_followed: targetUserId}
+      replacements: {user_id_follower: req?.userId || '', user_id_followed: targetUserId}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Using an empty string as a replacement for `user_id_follower` when `req.userId` is undefined could potentially lead to unexpected behavior or errors in your SQL query. It would be better to handle this case explicitly, perhaps by altering the query or its conditions based on whether `req.userId` is defined.

Without knowing the exact structure and requirements of your database and queries, I can't provide a specific code fix, but consider revising this line to ensure that it behaves correctly when `req.userId` is undefined.


<!-- This is an auto-generated comment by OSS CodeRabbit -->